cpp/ql/lib/semmle/code/cpp/ir/internal/CppType.qll

```diff
@@ -538,12 +536,14 @@ CppType getCanonicalOpaqueType(Type tag, int byteSize) {
 }
 
 /**
- * Gets a string that uniquely identifies an `IROpaqueType` tag. This may be different from the usual
- * `toString()` of the tag in order to ensure uniqueness.
+ * Gets a string that uniquely identifies an `IROpaqueType` tag. Using `toString` here might
+ * not be sufficient to ensure uniqueness, but suffices for our current debugging purposes.
+ * To ensure uniqueness `getOpaqueTagIdentityString` from `semmle.code.cpp.Print` could be used,
+ * but that comes at the cost of importing all the `Dump` classes defined in that library.
  */
 string getOpaqueTagIdentityString(Type tag) {
   hasOpaqueType(tag, _) and
-  result = getTypeIdentityString(tag)
+  result = tag.toString()
 }
```

Comment:
This is probably somewhat dubious. However, this still seems to suffice for our tests (which is the only place where this is used). I tried to factor this out and keep the old definition. Unfortunately this means factoring out all `getUniqueId` and related predicates, which if non-trivial.